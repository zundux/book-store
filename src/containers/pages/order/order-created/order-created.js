import React from "react";
import { connect } from "react-redux";
import { SUPPORT_EMAIL_ADDRESS } from "../../../../api";

export const OrderCreated = ({ orderId }) => (
  <div>
    {orderId && (
      <p>Order â„– <b>{orderId}</b> was created successfully</p>
    )}
    {!orderId && (
      <p>
        Something went wong.
        <br/>
        Please, contact us
        &nbsp;
        <a href={`mailto:${SUPPORT_EMAIL_ADDRESS}`}>
          {SUPPORT_EMAIL_ADDRESS}
        </a>
      </p>
    )}
  </div>
);

const mapStateToProps = ({ order }) => ({
  orderId: order.id
});

export default connect(
  mapStateToProps,
  null
)(OrderCreated);