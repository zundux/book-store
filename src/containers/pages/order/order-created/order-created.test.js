import React from "react";
import { OrderCreated } from "./order-created";

describe("Order created page", () => {

  it("Order created page should not show success message by default", () => {
    const orderId = 11;

    const component = shallow(
      <OrderCreated {...{
        orderId
      }}/>
    );
    matchSnapshot(component);
    // todo check page header
    // todo check page title
  });

  it("Order page should show error message if something went wrong", () => {
    const orderId = null;

    const component = shallow(
      <OrderCreated {...{
        orderId
      }}/>
    );
    matchSnapshot(component);
    // todo check page header
    // todo check page title
  });
});