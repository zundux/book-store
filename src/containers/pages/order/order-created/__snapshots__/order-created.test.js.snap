// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order created page Order created page should not show success message by default 1`] = `
<div>
  <p>
    Order № 
    <b>
      11
    </b>
     was created successfully
  </p>
</div>
`;

exports[`Order created page Order page should show error message if something went wrong 1`] = `
<div>
  <p>
    Something went wong.
    <br />
    Please, contact us  
    <a
      href="mailto:support@localhost"
    >
      support@localhost
    </a>
  </p>
</div>
`;
